src/app/logic/automatic-compilation.ts

```diff
@@ -14,6 +14,22 @@ export function setupAutomaticCompilationAndDownloadURL(): void {
   );
 }
 
+/**
+ * Get the worker to signal when compilation is successful.
+ */
+export function setupCompilationSuccessSignal(): void {
+  worker.onPackageCompileStart(
+    Comlink.proxy(() => {
+      compileSuccess.set(false);
+    })
+  );
+  worker.onPackageCompileSuccess(
+    Comlink.proxy(() => {
+      compileSuccess.set(true);
+    })
+  );
```

Comment:
So I don't believe the worker allows for multiple `onPackageCompileStart` callbacks, which means this _overwrites_ the existing `onPackageCompileSuccess` callback; namely, the one that sets up the download button.

The solution is to merge the logic here into `setupAutomaticCompilationAndDownloadURL()` and remove the `setupCompilationSuccessSignal()` function. There's no harm in setting the stores even if there are no listeners to the store's changes.

Reply:
Good catch